pkg/kubelet/cm/topologymanager/topology_manager.go

```diff
@@ -166,20 +176,14 @@ func (m *manager) calculateAffinity(pod v1.Pod, container v1.Container) Topology
 
 		// If hints is nil, insert a single, preferred any-socket hint into allProviderHints.
 		if hints == nil || len(hints) == 0 {
-			klog.Infof("[topologymanager] Hint Provider has no preference for socket affinity with any resource")
-			affinity, _ := socketmask.NewSocketMask()
-			affinity.Fill()
-			allProviderHints = append(allProviderHints, []TopologyHint{{affinity, true}})
+			klog.Infof("[topologymanager] Hint Provider has no preference for socket affinity with any resource skipping ...")
 			continue
 		}
 
 		// Otherwise, accumulate the hints for each resource type into allProviderHints.
 		for resource := range hints {
 			if hints[resource] == nil {
-				klog.Infof("[topologymanager] Hint Provider has no preference for socket affinity with resource '%s'", resource)
-				affinity, _ := socketmask.NewSocketMask()
-				affinity.Fill()
-				allProviderHints = append(allProviderHints, []TopologyHint{{affinity, true}})
```

Comment:
Removing this from here and above changes the semantics of the existing code slightly.
In both of these cases we are essentially creating a `defaultHint` of {full, true} (which will override the original default hint of {full, false} when hints are merged.

With your changes, the result of all resource types returning a `nil` will not result in {full, true} (as it should), but rather {full, false} because the default will never be overridden.

Reply:
In case I will put it back then in policy strict it will mean that we will failed to admit the pod (in case of one resource not having hints) . if that is the required behavior then I would put it back.  So I mean that the for best-effort it will be don't care and for strict it will mean don't admit.

Reply:
Yeah, we had this discussion before (somewhere other than this PR it seems, but I don't remember where). We concluded that the semantics encoded here are highly dependent on the `best-effort` policy.

As such, I actually pictured the abstraction you are building here to allow customization of the entire `calculateAffinity()` call and not just a replacement of the merge strategy.

Reply:
yes it was in the resource map. The reason I did it like this was that for best-effort I don't think it matter if the hint is {full, ture} or {full, false} (please correct me if I am wrong), but it does affect the strict policy. 
Customization of the calculateAffinity is a bit more complicated and unfortunately I on PTO until end of month.   I think the question is if this is good enough for current release. 
Anyway  adrianchiris "adrianc@mellanox.com" from my team will continue working this if required.

Reply:
strict policy should not prevent pod admission in case a resource does not care about affinity.
on the bright side, SR-IOV (network) and CPU resources do care about affinity.

Anyway, I still need to properly catch up with related PR, will try to do that today.

Reply:
I’d prefer it if the new Merge() call actually encapsulated what right now is called calculateAffinity(). It might lead to some amount of duplicated code, but my guess is that it will be less than we expect if we clean up the strict implementation a bit.

The current implementation of calculateAffinity() will then simply become the new Merge() call for the best effort policy, and a new (simplified) one can be written for strict.

Reply:
makes sense, i believe default hint generation should be per policy,
(or allow a policy to generate a default hint)
i will work on it next week.